### PR TITLE
Feautre: d3.js based Box Plot visualization.

### PR DESCRIPTION
**Pull request by matthew-sochor**
Adding d3js boxplot to re:dash visualizations.  Box plot code derived from: http://bl.ocks.org/mbostock/4061502

### CODE REVIEW

File: rd_ui/app/scripts/visualizations/boxplot.js [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    }
+  ]);
+  module.directive('boxplotRenderer', function() {
+    return {
+      restrict: 'E',
+      templateUrl: '/views/visualizations/boxplot.html',
+      link: function($scope, elm, attrs) {
+
+        
+        var loaddata = function(inputdata){
```
Comment (arikfr) [inline rd_ui/app/scripts/visualizations/boxplot.js]: this seems to be unused?

File: rd_ui/app/scripts/visualizations/boxplot.js [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+        '<boxplot-renderer ' +
+        'options="visualization.options" query-result="queryResult">' +
+        '</boxplot-renderer>';
+
+      var editTemplate = '<boxplot-editor></boxplot-editor>';
+      var defaultOptions = {
+        boxplotColName: 'boxplot',
+        rowNumber: 1,
+        targetRowNumber: 1
+      };
```
Comment (arikfr) [inline rd_ui/app/scripts/visualizations/boxplot.js]: it looks like you don't use these settings?

**Comment by arikfr** [comment]
Thanks! Looks good, but please see my comments.

File: rd_ui/app/scripts/visualizations/boxplot.js (hunk 1 above)
Comment (matthew-sochor) [inline rd_ui/app/scripts/visualizations/boxplot.js]: You are correct! It was used while I was developing and I guess it managed to hang around. 
Fixed in 0b5dce0

File: rd_ui/app/scripts/visualizations/boxplot.js (hunk 2 above)
Comment (matthew-sochor) [inline rd_ui/app/scripts/visualizations/boxplot.js]: Yes they are unused.  Fixed in 0b5dce0

File: rd_ui/app/scripts/visualizations/boxplot.js [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+            while (d[++i] < q1 - iqr);
+            while (d[--j] > q3 + iqr);
+            return [i, j];
+          };
+        };
+
+        $scope.$watch('[queryResult && queryResult.getData(), visualization.options]', function () {
+          
+          var data = $scope.queryResult.getData();
+          var colName = $scope.visualization.options.colName;
```
Comment (arikfr) [inline rd_ui/app/scripts/visualizations/boxplot.js]: This looks unused too.

Comment (matthew-sochor) [inline rd_ui/app/scripts/visualizations/boxplot.js]: Fixed in 2feb8b8

**Comment by arikfr** [comment]
Merged. Thanks!

**Comment by matthew-sochor** [comment]
Excellent, thanks :+1:

**Pull request merged**